lms/static/scripts/frontend_apps/components/StudentSelector.js

```diff
@@ -86,30 +93,63 @@ export default function StudentSelector({
         >
           {options}
         </select>
-        <Icon classes="StudentsSelector__students-icon" name="caretDown" />{' '}
+        <Icon
+          classes="absolute top-0.5 right-3 pointer-events-none text-grey-4"
+          name="caretDown"
+        />{' '}
       </span>
     );
   };
 
   return (
-    <div className="StudentSelector">
-      <button
-        className="StudentSelector__change-student"
-        aria-label="previous student"
-        disabled={!hasPrevView}
-        onClick={onPrevView}
+    <div
+      className={classNames(
+        // Narrower widths: label above field
+        'flex flex-col gap-1',
+        // Wider widths: label to left of field
+        'xl:flex-row xl:gap-3 xl:items-center'
+      )}
+    >
+      <label
+        className={classNames(
+          'flex-grow font-medium text-sm leading-none',
+          'xl:text-right'
+        )}
+        data-testid="student-selector-label"
       >
-        <Icon classes="StudentSelector__svg" name="arrowLeft" />
-      </button>
-      {buildStudentList()}
-      <button
-        className="StudentSelector__change-student"
-        aria-label="next student"
-        disabled={!hasNextView}
-        onClick={onNextView}
-      >
-        <Icon classes="StudentSelector__svg" name="arrowRight" />
-      </button>
+        {selectedStudentIndex >= 0 ? (
+          <span>
+            Student {`${selectedStudentIndex + 1} of ${students.length}`}
+          </span>
+        ) : (
+          <span>{`${students.length} Students`}</span>
+        )}
+      </label>
+      <div className="flex flex-row">
+        <IconButton
+          classes={classNames(
+            'px-3 rounded-r-none',
```

Comment:
Is the border radius (`rounded-r-none`) necessary?

Reply:
Yes, turning off border radius on the right side .`IconButton`'s styling sets border-radius. I'll double-check, however, and add commenting since it gave you pause (commenting is good!).